inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/special/ExecutorContextPropagationSensor.java

```diff
@@ -0,0 +1,55 @@
+package rocks.inspectit.oce.core.instrumentation.special;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.dynamic.DynamicType;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.stereotype.Component;
+import rocks.inspectit.oce.bootstrap.Instances;
+import rocks.inspectit.oce.core.config.model.instrumentation.InstrumentationSettings;
+import rocks.inspectit.oce.core.instrumentation.context.ContextManagerImpl;
+
+import java.util.concurrent.Executor;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+
+@Component
+@DependsOn(ContextManagerImpl.BEAN_NAME)
+@Slf4j
+public class ExecutorContextPropagationSensor implements SpecialSensor {
+    @Override
+    public boolean shouldInstrument(TypeDescription type, InstrumentationSettings settings) {
+        val matcher = isSubTypeOf(Executor.class)
```

Comment:
can we extract the byte buddy matcher to the `static` field, this seems to be constant and we are always recreating it?